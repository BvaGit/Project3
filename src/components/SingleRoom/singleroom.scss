@import '../../utils/variables.scss';

.singleroom{
    background:rgb(214, 214, 214);
    margin: 3px;
    padding: $padding_md;
    @include flex(_, space-between,_);
    &__info{
        @include flex(_,_,_);
    }
    &__avatar{
        width: 25px;
        border: $border solid 1px;
        border-radius: 50%;
    }
    &__roomName{
        align-self: center;
        margin-left: $margin_sm;
        font-family: 'Noto Sans KR', sans-serif;
    }
    &__messages{
        @include flex(_,_,_);
    }
}
.singleroom:hover{
    -webkit-transform: scale(1.005);
    -ms-transform: scale(1.005);
    transform: scale(1.005);
    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
    transition-duration: 0.3s;
}