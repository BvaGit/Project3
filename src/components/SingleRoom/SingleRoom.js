import React from "react";

import MessageCounter from "../MessageCounter";
import MentionPin from "../MentionPin";

import "./singleroom.scss";

const SingleRoom = ({ room, setActiveRoom, getChatMessages }) => {
  const handleClick = () => {
    setActiveRoom(room.chat_id);
    getChatMessages(room.chat_id);
  };
  return (
    <div className="singleroom" onClick={handleClick}>
      <div className="singleroom__info">
        <img
          className="singleroom__avatar"
          src="../../../public/assets/images/1-36-1024x1024.jpg"
        />
        <div className="singleroom__roomName" key={room.chat_id}>
          {room.name}
        </div>
      </div>
      <div className="singleroom__messages">
        <MentionPin />
        <MessageCounter />
      </div>
    </div>
  );
};

export default SingleRoom;
