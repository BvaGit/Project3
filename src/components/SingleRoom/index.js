import { connect } from "react-redux";

import SingleRoom from "./SingleRoom";

import { getChatMessages } from "../../store/rooms/asyncActions";

const mapDispatchToProps = (dispatch) => ({
  getChatMessages: (chatId) => dispatch(getChatMessages(chatId)),
});

export default connect(null, mapDispatchToProps)(SingleRoom);
