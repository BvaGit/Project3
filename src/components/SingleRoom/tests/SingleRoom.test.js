import React from 'react';
import { shallow, mount } from 'enzyme';
import SingleRoom from '../SingleRoom';


describe('SingleRoom', () => {
    let props;
    it("Should match snapshot", () => {
        props={room:{name:'global',chat_id:1}}
        const component = shallow(<SingleRoom {...props}/>)
        expect(component.html()).toMatchSnapshot()
    })
})