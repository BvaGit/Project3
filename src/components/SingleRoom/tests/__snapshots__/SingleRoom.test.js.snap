// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleRoom Should match snapshot 1`] = `"<div class=\\"singleroom\\"><div class=\\"singleroom__info\\"><img class=\\"singleroom__avatar\\" src=\\"../../../public/assets/images/1-36-1024x1024.jpg\\"/><div class=\\"singleroom__roomName\\">global</div></div><div class=\\"singleroom__messages\\"><div class=\\"mention-pin\\">@</div><div class=\\"message-counter mention-pin\\">99..</div></div></div>"`;
